task-sdk/docs/api.rst

```diff
@@ -0,0 +1,159 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+airflow.sdk API Reference
+=========================
+
+This page documents the full public API exposed in Airflow 3.0+ via the Task SDK python module.
+
+If something is not on this page it is best to assume that it is not part of the public API and use of it is entirely at your own risk
+-- we won't go out of our way break usage of them, but we make no promises either.
+
+.. :py:module: airflow.sdk
+
+Defining DAGs
+-------------
+.. autoapiclass:: airflow.sdk.DAG
+
+.. autoapifunction:: airflow.sdk.dag
+
+Decorators
+----------
+.. autoapifunction:: airflow.sdk.task
+
+.. autoapifunction:: airflow.sdk.task_group
+
+.. autoapifunction:: airflow.sdk.setup
+
+.. autoapifunction:: airflow.sdk.teardown
+
+.. autofunction:: airflow.sdk.task
+.. autofunction:: airflow.sdk.setup
+.. autofunction:: airflow.sdk.teardown
+.. autofunction:: airflow.sdk.asset
+
+Utilities
+---------
+.. autoapifunction:: airflow.sdk.chain
+
+.. autoapifunction:: airflow.sdk.chain_linear
+
+.. autoapifunction:: airflow.sdk.cross_downstream
+
+.. autoapifunction:: airflow.sdk.literal
+
+.. autoapifunction:: airflow.sdk.__version__
+.. autofunction:: airflow.sdk.__version__
```

Comment:
a) Why do we have this twice?
b) it looks wrong:
<img width="1261" alt="Screenshot 2025-06-03 at 10 27 13" src="https://github.com/user-attachments/assets/0ef7163a-5b18-4607-b270-cea43ab3da88" />

c) do we even want to document this? I think not.

Reply:
The rendered docs  for publishing will show the version selector

<img width="1122" alt="Screenshot 2025-06-03 at 14 24 54" src="https://github.com/user-attachments/assets/e9b3d5c6-e902-4b63-b2d7-3ace7097917a" />

so we don't need to try and show the version here.